packages/react-server-dom-webpack/src/ReactFlightWebpackNodeRegister.js

```diff
@@ -57,7 +57,7 @@ module.exports = function register() {
     },
   };
 
-  (require: any).extensions['.client.js'] = function(module, path) {
+  Module._extensions['.client.js'] = function(module, path) {
```

Comment:
I made this change only because of tests. jest annoyingly overrides `require.extensions` for every module so there's no way to inspect this otherwise since it's not shared.

So I moved it to use `Module._extensions` which is more of an internal implementation detail but both are "deprecated" but effectively stable and no reasonable alternative.

We currently override `_resolveFilename` anyway but we can get rid of that when we change to the new conventions.